### DISCUSSION TITLE
How in resource table with several long text fields hide part of text and status make switcher ?

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#13529] [Help] [answered]
### Package

Table builder

### Package Version

^3.2-stable

### How can we help you?

In laravel 10 / filamentphp 3 app in resource table I have a table with several long text and status "Not Viewed" .
I search if there is a way  :

a) To show some part of  long text field(adding some button like "...")
b) Clicking on that "..." button to show this full long text in modal popup window
c) Some switcher how set status into "Viewed"  without opening resource edit page for any row

Are there some examples/pugins for this ?

**Comment by dissto** [comment, marked as answer]
a) [Limiting text length](https://filamentphp.com/docs/3.x/tables/columns/text#limiting-text-length)
b) [Running actions](https://filamentphp.com/docs/3.x/tables/columns/getting-started#running-actions)
c) [Toggle column](https://filamentphp.com/docs/3.x/tables/columns/toggle) / [Select column](https://filamentphp.com/docs/3.x/tables/columns/select)

🤔

**Comment by lukasinko** [comment]
@dissto I'm curious how to make the "show full long text in modal popup window", could you give me some hints ? I was reading the section "Running actions", but i don't know how to show the text from the column cell. Thank you in advance.

**Reply by lukasinko** [reply]
So i've finally figured it out. The most important function is `default`. Full example:

```
Tables\Columns\TextColumn::make('note')
                    ->wrap()
                    ->words(10)
                    ->action(Tables\Actions\Action::make('note')->form([
                        Forms\Components\Textarea::make('note')
                        ->default(function($record){
                            return $record->note;
                        })
                        ->autosize()
                    ]))
```
